src/utils/formatBalance.ts

```diff
@@ -30,3 +32,25 @@ export const formatNumber = (number: number, minPrecision = 2, maxPrecision = 2)
   }
   return number.toLocaleString(undefined, options)
 }
+
+/**
+ * Method to format the display of wei given an ethers.BigNumber object
+ * Note: does NOT round
+ */
+export const formatBigNumber = (number: ethers.BigNumber, displayDecimals = 18, decimals = 18) => {
+  const balance = formatUnits(number, decimals)
+  const [leftSide, rightSide] = balance.split('.')
+
+  let charCount = displayDecimals
+
+  if (leftSide.length > 0) {
+    charCount += leftSide.length
+  }
+
+  // Add 1 for the decimal point
+  if (rightSide.length > 0) {
+    charCount += 1
+  }
```

Comment:
I'm a bit confused by all this conditions:
- `leftSide.length` will always be more than 0, no? Seems that no need to check for that
- `rightSide.length` also will be always > 0 unless token `decimals` is 0. In that case whole `rightSide` will be `undefined` and `undefined.length` will throw error.